### ISSUE TITLE
Auto-seeding a node is dangenrous

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylladb#15836]
When the `seeds` option is missing (or misspelled, commonly as singular `seed`), Scylla just happily uses the localhost address as seed, allowing the node to bootstrap itself as a new, single node cluster.

See https://github.com/scylladb/scylladb/blob/6c90d166cc14cbdfb39f8f6e7bd8c32608531e87/init.cc#L43-L45

I suggest we do that only in developer mode, otherwise the node should refuse to boot with an error indicating that no seeds were configured.

Open questions:
1. Should we also accept `seed`, if `seeds` is not provided?
2. Once the node bootstrapped. I think we can just ignore `seeds` and simply use the known system.peers as seeds.  This will simplify some topology changes that have special considerations around seed nodes, and the need to maintain `seeds` in scylla.yaml if a seed node address needs changing, or it needs to be decommissioned.

Cc @asias @kbr-scylla @tgrabiec

**Comment by bhalevy** [comment]
https://github.com/scylladb/scylla-enterprise/issues/3523#issuecomment-1774826980

> Related issue: https://github.com/scylladb/scylladb/issues/14448
> and fix: https://github.com/scylladb/scylladb/pull/15004
>
> tl;dr in master and 5.4. with Raft enabled, nodes from separate clusters will refuse to gossip with each other because of different group 0 IDs.

**Comment by bhalevy** [comment]
Despite the above, I think we should fix this issue, even if for backporting purposes.

**Comment by kbr-scylla** [comment]
> Once the node bootstrapped. I think we can just ignore seeds and simply use the known system.peers as seeds.

Seeds are still needed for IP address changes. If all nodes change their IPs the only way for the cluster to discover itself back is through seeds
